jujugui/static/gui/src/app/components/user-profile/model-list/model-list.js

```diff
@@ -108,6 +110,117 @@ YUI.add('user-profile-model-list', function() {
     },
 
     /**
+      Display the confirmation for destroying a model.
+
+      @method _displayConfirmation
+      @param {Object} model the model to destroy. A model should at least have
+                            a name and uuid.
+    */
+    _displayConfirmation: function(model) {
+      this.setState({modelToBeDestroyed: model});
+    },
+
+    /**
+      Generate the confirmation for destroying a model.
+
+      @method _displayConfirmation
+      @return {Object} the confirmation component.
+    */
+    _generateConfirmation: function() {
+      const model = this.state.modelToBeDestroyed;
+      const addNotification = this.props.addNotification;
+      if (!model) {
+        return;
+      }
+      if (model.isAdmin) {
+        addNotification({
+          title: 'Cannot destroy model',
+          message: 'The controller model cannot be destroyed.',
+          level: 'error'
+        });
+        return;
+      }
+      const buttons = [{
+        title: 'Cancel',
+        action: this._displayConfirmation.bind(this, null),
+        type: 'base'
+      }, {
+        title: 'Destroy',
+        action: this._destroyModel,
+        type: 'destructive'
+      }];
+      const message = `Are you sure you want to destroy ${model.name}?`
+        + ' All the applications and units included in the model will be'
+        + ' destroyed. This action cannot be undone.';
+      return (
+        <juju.components.ConfirmationPopup
+          buttons={buttons}
+          message={message}
+          title="Destroy model" />);
+    },
+
+    /**
+      Makes a request of the controller to delete a selected model.
+
+      @method _destroyModel
+    */
+    _destroyModel: function() {
+      const model = this.state.modelToBeDestroyed;
+      const uuid = model.uuid;
+      // Hide the confirmation popup.
+      this._displayConfirmation(null);
+      const xhr = this.props.destroyModels([uuid], this._destroyModelCallback);
+      this.xhrs.push(xhr);
+    },
+
+    /**
+      Callback for the controller delete model call.
+
+      @method _destroyModelCallback
+      @param {String} err The error from the request, or null.
+      @param {Array} modelList The list of models.
+    */
+    _destroyModelCallback: function(err, results) {
+      const addNotification = this.props.addNotification;
+      // Handle global errors.
+      if (err) {
+        addNotification({
+          title: 'Model destruction failed',
+          message: 'The model failed to be destroyed: ' + err,
+          level: 'error'
+        });
+        return;
+      }
+      // Ignore all but the first UUID.
+      const uuid = Object.keys(results)[0];
```

Comment:
I'm not really sure about this implementation. It seems like looping through all of them would be just as much work and would allow us to easily add the ability to delete more at once in the future.